Sources/VitalHealthKit/HealthKit/VitalHealthKitClient.swift

```diff
@@ -93,18 +84,92 @@ public extension VitalHealthKitClient {
     
     public init(
       autoSync: Bool = false,
+      backgroundUpdates: Bool = false,
       logsEnable: Bool = true
     ) {
       self.autoSync = autoSync
-      // More testing is required before exposing this feature
-      self.backgroundUpdates = false
+      self.backgroundUpdates = backgroundUpdates
       self.logsEnable = logsEnable
     }
   }
 }
 
 extension VitalHealthKitClient {
   
+  private func checkBackgroundUpdates(isBackgroundEnabled: Bool, resources: [VitalResource]) {
+    guard isBackgroundEnabled else { return }
+    guard resources.isEmpty == false else { return }
+    
+    /// If it's already running, cancel it
+    self.backgroundDeliveryTask?.cancel()
+
+    self.backgroundDeliveryTask = Task(priority: .high) {
```

Comment:
This would not satisfy the background delivery requirement from HealthKit, on having all your `HKObserverQuery` configured before `application(_:didFinishLaunchingWithOptions:)` ends.

> As soon as your app launches, HealthKit calls the update handler for any observer queries that match the newly saved data. If you plan on supporting background delivery, set up all your observer queries in your app delegate’s [application(_:didFinishLaunchingWithOptions:)](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622921-application) method. [...]

In fact, there is no way[1] under Swift Concurrency to satisfy the requirement. Even if you create an unstructured task bound to the MainActor (via `Task { @MainActor in }`), the task will still have to wait until the next main queue drain to be executed (at the end of the current runloop iteration), which is way past the `didFinishLaunchingWithOptions` message already.

([1] There might be a new Swift Concurrency feature in the future to help with this, but it is currently a behaviour private to the Standard Library)

What you should consider is to run `backgroundObservers(for:)` synchronously to create your HKObserverQuery-fed `AsyncStream`, either before or during `application(_:didFinishLaunchingWithOptions:)`.

Since an `AsyncStream` is guaranteed to "start" (invoking the builder closure + the value buffering) **as soon as** the initializer returns, your `HKObserverQuery`s would have been definitively instantiated and registered with the `HKHealthStore` before `backgroundObservers(for:)` returns.

You are then safe to pass this active `AsyncStream` into this unstructured Task, both to setup the async iteration, and to carry out the actual data processing on the cooperative thread pool.

https://developer.apple.com/documentation/healthkit/hkhealthstore/1614175-enablebackgrounddelivery